src/backend/distributed/commands/drop_distributed_table.c

```diff
@@ -38,6 +38,8 @@ master_drop_distributed_table_metadata(PG_FUNCTION_ARGS)
 	char *schemaName = text_to_cstring(schemaNameText);
 	char *tableName = text_to_cstring(tableNameText);
 
+	EnsureSchemaNode();
```

Comment:
We already have a check in `create_reference_table`. I didn't understand what is the relation between `master_drop_distributed_table_metadata` and `create_distributed_table` though.